src/js/popup.js

```diff
@@ -116,13 +170,17 @@ function loadTemplateEditor (callback = false) {
         // Populate the add default template dropdown - excluding any templates already loaded.
         chrome.runtime.sendMessage({JDTIfunction: 'fetchDefault'}, function (response) {
             if (response.status === 'success') {
+                var bgPage = chrome.extension.getBackgroundPage();
                 var defaultTemplates = response.data.templates;
 
-                // Remove default templates from dropdown list if already added.
-                $.each(dropdownExcludeList, function (index, issueTypeField) {
-                    $.each(defaultTemplates, function (name, template) {
-                        if (issueTypeField === template['issuetype-field']) {
-                            delete defaultTemplates[name];
+                // Remove default templates from dropdown list if a template for that (issue type, projects) combination already exists.
+                $.each(templatesArray, function (index, excludeTemplate) {
```

Comment:
I ended up using Handlebars which is specifically for templating. Lodash seems a little overkill.